### DISCUSSION TITLE
Song repetition in mysql

### DISCUSSION THREAD

**Discussion by leandro-darchivio** [ampache/ampache#3691] [General]
Good afternoon everyone! I have problems because when uploading new songs to ampache and reviewing the "song" table of the mysql database I see that the songs are repeated when uploaded, 2, 3 or 4 times, how can I solve it when the error has occurred in at least 500 songs? thank you
![Sin título](https://github.com/ampache/ampache/assets/64814414/9eca3313-8aae-46ee-9977-ca61f8cf3c76)

**Comment by lachlan-00** [comment]
I'm going to add some more songs tomorrow. There is supposed to be lots of lookups and pre-checks before inserting songs.

What file system is this media drive?

**Comment by leandro-darchivio** [comment]
is a screenshot of the "song" table displayed with "admin". I see that the same thing happens to many people, and from the /bin/cli command I don't know if you can verify the repetition of files and delete them

**Reply by lachlan-00** [reply]
does your table have the song_UN constraint?
![image](https://github.com/ampache/ampache/assets/1305249/e68b52bb-3cc3-49cb-a5a6-1f5dc7b5a789)

it's not meant to allow dupes for file at all.

try copy & paste of the file column into a comment with the `code` tags
 
they look the same but maybe they aren't

**Reply by lachlan-00** [reply]
also double check that they're actually dupes with this query

```
SELECT `file`, COUNT(*) AS `count`
FROM `song`
GROUP BY `file`
HAVING COUNT(*) > 1;
```
